editor/src/clj/editor/app_view.clj

```diff
@@ -1215,14 +1227,23 @@
                                (when (handle-build-results! workspace render-build-error! build-results)
                                  (when (or engine skip-engine)
                                    (show-console! main-scene tool-tab-pane)
-                                   (launch-built-project! project engine project-directory prefs web-server false)))))))
+                                   (launch-built-project! project engine project-directory prefs web-server false workspace)))))))
 
 (handler/defhandler :build :global
   (enabled? [] (not (build-in-progress?)))
   (run [project workspace prefs web-server build-errors-view debug-view main-stage tool-tab-pane]
     (debug-view/detach! debug-view)
     (build-handler project workspace prefs web-server build-errors-view main-stage tool-tab-pane)))
 
+(handler/defhandler :set-instance-count :global
+  (enabled? [] true)
+  (run [prefs user-data workspace]
+       (let [count (:instance-count user-data)]
+         (prefs/set-prefs prefs (prefs/make-project-specific-key "instance-count" workspace) count)))
+  (state [prefs user-data workspace]
+         (= (:instance-count user-data)
+            (prefs/get-prefs prefs (prefs/make-project-specific-key "instance-count" workspace) 1))))
```

Comment:
Do I need atom here? I'm not sure because I saw both implementation with and without atom.